recipes-core/initrdscripts/files/ni_provisioning.common

```diff
@@ -374,16 +374,12 @@ elif [ "$ARCH" = "x86_64" ]; then
             mount -o rw /dev/niboot/niboota /mnt/efi-auto/niboota
 
             local img_written=false
-            for imgname in "minimal-nilrt-bundle-image-x64.tar.bz2" "lvcomms-nilrt-bundle-image-x64.tar.bz2"; do
-                local imgpath="/mnt/efi-auto/bundle/$imgname"
+            for imgpath in /mnt/efi-auto/bundle/*.tar.bz2; do
```

Comment:
Under normal circumstances, there is only ever one image in the bundle - hence why both I and the previous author `break` out after the first loop. Both of our stanzas are just using the `for... do` loop to parse the first element from the wildcard expansion or inline element list respectively.

I could replace it with some explicit lines to parse out the first element. The only substantial difference would be that, as I have it written now, if there isn't an image file, the script will exit with a bash syntax error (as the wildcard expansion becomes ""). I think the ROI on using a more explicit codeblock is pretty low here though.

Reply:
I was wondering if the script should return an error if there were more than one image, instead of silently accepting the first in the list. Leaving it to you, feel free to drop.

Reply:
I think it's pretty unlikely that we'll accidentally include more than one .tar.bz2 in the `bundle/` directory. To do that, we would need to intentionally modify the `bundle.bbclass` - which is owned by meta-rauc - and then also provide some alternate content in the bundle recipe. Maybe we'll feel differently about it once we start development, but I think just using a `for...do` loop is sufficient for this patchset.